### ISSUE TITLE
Unhardcode everything and split behaviors into digestible scene structure

### ISSUE THREAD

**Issue by Crystalwarrior** [coppolaemilio/dialogic#45]
Currently, Dialogic interface is hard-coded and difficult to expand with extra systems any developer would like to have (for example, my attempts to implement an Ace Attorney cross-examination system are hell right now.)
![image](https://user-images.githubusercontent.com/3470436/106488005-4fcaf200-64c4-11eb-995b-bc931ff88953.png)
*example of current hard-coded nature*

My proposal is to expand and unhardcode the system in such a way that it's split into several digestible scenes, all of which can be replaced and overriden by developers while Dialogic still provides out-of-the-box default versions of the behaviors. This way, beginners will have an immediately working interface - what Dialogic is very good at right now - and also provide an alternative for advanced users who want to use the backend of Dialogic while handling the frontend themselves.
![image](https://user-images.githubusercontent.com/3470436/106488183-7c7f0980-64c4-11eb-9abe-fda0d2a9273f.png)
*example of a dynamic and nice version*

In simple terms, the parser node is the big focus of this issue report. A very good principle to hold for this is as such:
![image](https://user-images.githubusercontent.com/3470436/106487613-e5b24d00-64c3-11eb-92a7-e5d66a77e308.png)
*example how a parser node only acts as an intermediary for anything connected to it to interpret the data*

This approach will also solidify Dialogic's position in the Godot plugin scene and might even make it the #1 go-to plugin for dialogue interfaces 🤔
